communicator/vmturbocommunicator/server_message_handler.go

```diff
@@ -268,17 +278,41 @@ func (handler *MesosServerMessageHandler) HandleAction(serverMsg *comm.Mediation
 func (handler *MesosServerMessageHandler) NewMesosProbe(previousUseMap map[string]*util.CalculatedUse) (*util.MesosAPIResponse, error) {
 	fullUrl := "http://" + handler.meta.MesosIP + ":" + handler.meta.MesosPort + "/state"
 	glog.V(4).Infof("The full Url is ", fullUrl)
+
 	req, err := http.NewRequest("GET", fullUrl, nil)
 
+	if handler.meta.DCOS {
+		req.Header.Add("content-type", "application/json")
+		req.Header.Add("authorization", "token="+handler.meta.Token)
+	}
+
 	glog.V(4).Infof("%+v", req)
 	client := &http.Client{}
 	resp, err := client.Do(req)
+
 	if err != nil {
-		glog.Errorf("Error getting response: %s \n", err)
+		glog.V(3).Infof("Error in GET request to mesos master: %s\n", err)
+		return nil, err
+	}
+	// Get token if response if OK
+	if resp.Status == "" {
+		glog.V(3).Infof("Empty response status\n")
 		return nil, err
```

Comment:
When the res.Status is empty, isn't this an error? And I believe the err returned is always nil.